src/action-registry/action-registry.controller.spec.ts

```diff
@@ -0,0 +1,180 @@
+import { PrismaModule } from '@/prisma/prisma.module';
+import { Test } from '@nestjs/testing';
+import { Action } from '@prisma/client';
+import { solidityKeccak256 } from 'ethers/lib/utils';
+import { ActionRegistryController } from './action-registry.controller';
+import { ActionRegistryService } from './action-registry.service';
+import { ActionDto, CreateActionDto, UpdateActionDto } from './dto';
+
+describe('ActionRegistryController', () => {
+  let controller: ActionRegistryController;
+  let service: ActionRegistryService;
+
+  beforeEach(async () => {
+    const moduleRef = await Test.createTestingModule({
+      controllers: [ActionRegistryController],
+      providers: [ActionRegistryService],
+      imports: [PrismaModule],
+    }).compile();
```

Comment:
Instead of providing a real implementation of service use `mock` function from `jest-mock-extended`. This will allow you to run unit tests without having a dependency from database connection.

Something like this:

```suggestion
    const moduleRef = await Test.createTestingModule({
      controllers: [ActionRegistryController],
      providers: [
        {
          provide: ActionRegistryService,
          useValue: mock<ActionRegistryService>(),
        },
      ],
    }).compile();
```

And you could type `service` as `MockProxy<ActionRegistryService>` so you could use mock functions instead of spies:
```ts
service.getManyByGameId.mockResolvedValue(result);
```

You could also do some stubs testing like:
```ts
expect(service.getManyByGameId).toHaveBeenCalledWith(arg1, arg2, ...);
```

Reply:
Solved in https://github.com/ggQuest/api-v2-nest/pull/155/commits/c25239310136080a0798c78878ea354e132c1d31